theoplayer/how-to-guides/01-ads/13-custom-ssai-integration.md

```diff
@@ -0,0 +1,158 @@
+    skipOffset = 5,
+    id = "first_ad_id",
+    duration = 10,
+    clickThrough = first_ad_click_through_url,
+    resourceURI = first_ad_resource_url
+)
+val firstAd = controller.createAd(firstAdInit, adBreak)
+
+val secondAdInit = AdInit(
+    type = "linear",
+    skipOffset = 5,
+    id = "second_ad_id",
+    duration = 10,
+    clickThrough = second_ad_click_through_url,
+    resourceURI = second_ad_resource_url
+)
+val secondAd = controller.createAd(secondAdInit, adBreak)
+```
+
+#### Playing an Ad
+
+When an Ad begins the playback, the controller must inform the player by calling `beginAd(ad: Ad)`, this will trigger the relevant AdBreak and Ad events.
+
+Additionally, during the playback of the Ad, calling `updateAdProgress(ad: Ad, progress: Double)` is needed to keep the player in sync with the progress.
+
+Finally, when the playback of the Ad is completed, `endAd(ad: Ad)` should be called to notify about it.
+
+### CustomSsaiDescriptionSerializer
+
+To hold the configuration parameters specific to your integration, you can create a new class that inherits from `CustomSsaiDescription`.
+This allows it to be set on a `TypedSource` through [`TypedSource.Builder.ssai()`](<pathname:///theoplayer/v8/api-reference/android/com/theoplayer/android/api/source/TypedSource.Builder.html#ssai(com.theoplayer.android.api.source.ssai.SsaiDescription)>).
+Then, register a `CustomSsaiDescriptionSerializer` using the integration ID and provide a way to serialize/deserialize the new `CustomSsaiDescription`.
+
+Below is an example of a [YoSpace integration](https://github.com/THEOplayer/android-connector/tree/master/connectors/yospace) showing `YospaceSsaiDescription` extending `CustomSsaiDescription`:
+
+```kotlin
+data class YospaceSsaiDescription constructor(
+    /**
+     * The type of the requested stream.
+     *
+     * Default: [YospaceStreamType.LIVE]
+     */
+    val streamType: YospaceStreamType = YospaceStreamType.LIVE,
+    /**
+     * Custom properties to set when initializing the Yospace session.
+     */
+    val sessionProperties: Session.SessionProperties = Session.SessionProperties()
+) : CustomSsaiDescription() {
+    override val customIntegration = YospaceConnector.INTEGRATION_ID
+}
+```
+
+And then, the implementation of the `CustomSsaiDescriptionSerializer` specific for YoSpace:
+
+```kotlin
+class YospaceSsaiDescriptionSerializer : CustomSsaiDescriptionSerializer {
+    override fun fromJson(json: String): YospaceSsaiDescription {
+        return Json.decodeFromString(json)
+    }
+
+    override fun toJson(value: CustomSsaiDescription): String {
+        return Json.encodeToString(value)
+    }
+}
+```
+
+Finally, registering the `YospaceSsaiDescriptionSerializer`:
+
+```kotlin
+CustomSsaiDescriptionRegistry.register(YospaceConnector.INTEGRATION_ID, YospaceSsaiDescriptionSerializer())
+```
+
+> **Note**: Registering the CustomSsaiDescriptionSerializer should happen before registering the ServerSideAdIntegrationFactory.
```

Comment:
This would look prettier with an [admonition](https://docusaurus.io/docs/markdown-features/admonitions). 😉 

```suggestion
:::note

Registering the `CustomSsaiDescriptionSerializer` should happen before registering the `ServerSideAdIntegrationFactory`.

:::
```